### ISSUE TITLE
Cypress is stalling in particular test endlessly in npx cypress run mode, I can't get any report or time out issue

### ISSUE THREAD

**Issue by aancelin** [cypress-io/cypress#30097]
### Current behavior

When I run the tests in `npx cypress run` mode, it hangs in the second test case and never terminates the execution. I reproduced the issue on both Jenkins and my PC. The codes are fine, I don't know what is causing the problem. I noticed the address bar was blank too.

![image](https://github.com/user-attachments/assets/2e42ac64-01a0-4320-a3cc-bf3b935dcb54)

### Desired behavior

It should complete the test execution.

### Test code to reproduce
```js
   loginAthena(email, password){
        cy.visit(Cypress.env('host'));
        Wait.waitForPageLoad();
        this.txtEmail.type(email);
        this.txtPassword.type(password);
        this.btnLogin.click();
        Wait.waitForPageLoadBySec(30);
        cy.get('div').then($el => {
            if($el.find('.loginBtn').length > 0) {
                cy.log("===> Retrying...")
                this.btnLogin.click();
            }
        })
        HomePage.captionCarousel.should('exist')
        cy.log("===> Successfully logged in")
        Wait.waitForPageLoadBySec(30);
        cy.wait(3000)
    }
```
### Cypress Version

13.13.1

### Node version

14.21.3

### Operating System

win 11, jenkins linux 5.3.0

### Other

I reproduced the issue in Cypress open mode on the first try but I overcame the problem by clicking the rerun button. I don't have this option in headless mode.

**Comment by jennifer-shehane** [comment]
@aayazsin Is this a new test you've never run before? Or is this a test that was working and is now hanging? If so, what changed since the time it was working?

**Comment by aancelin** [comment]
In `cypress open` mode, the tests are passing, we started to run tests recently on Jenkins so this problem has been present since the first time we ran particular tests in Jenkins.

**Comment by MikeMcC399** [comment]
@aayazsin 
You should also make sure that you are running a supported version of Node.js, which means a minimum of `18.x` and preferably the LTS version `20.x`.

**Comment by aancelin** [comment]
ok thanks for the info let me try and get back to you...

**Comment by aancelin** [comment]
@MikeMcC399, @jennifer-shehane, I installed node version v20.17.0 but the issue is still present.

**Comment by aancelin** [comment]
Hi again, this issue is a blocker for our company, and it isn't feasible to continue with Cypress this way. It would be great if you could suggest a solution to this problem.

**Comment by jennifer-shehane** [comment]
@aayazsin Can you provide a reproducible example? Also ensure you are meeting our recommended machine/memory requirements to get optimal performance: https://docs.cypress.io/guides/getting-started/installing-cypress#Hardware

**Comment by aancelin** [comment]
@jennifer-shehane I don't think so, my pc has an i7 2.20ghz CPU with 64GB memory and Jenkins is running on c5.2xlarge server. Here is a video related to the issue:
[cypress.zip](https://github.com/user-attachments/files/16862667/cypress.zip)

**Comment by jennifer-shehane** [comment]
@aayazsin This is super weird. Could you run Cypress in [debug mode](https://on.cypress.io/troubleshooting#Print-DEBUG-logs) mode and print the entire set of logs here?

Ideally having the code to run on our side would allow us to track down what is causing this to get stuck.

**Comment by aancelin** [comment]
@jennifer-shehane, here is what I found in the log: 
![image](https://github.com/user-attachments/assets/4e9a620c-bb96-4b23-9902-7cffac885ca1)
[develop-auto-nonprod.com-1725539315338.log](https://github.com/user-attachments/files/16893487/develop-auto-nonprod.com-1725539315338.log)

**Comment by aancelin** [comment]
@jennifer-shehane, may I inquire if there have been any updates regarding the resolution of this bug?
